agere2015-akka-js/agere-akkajs.tex

```diff
@@ -147,20 +150,40 @@
 
 \section{Introduction}
 
-In recent times, there has been a significant pique of interest regarding JavaScript. The language, originally developed by Brendan Eich, while working at Netscape, and once confined to browsers, is now being used to power a wide variety of projects, ranging from server side applications \cite{web:nodejs} to hardware \cite{web:tessel} to robots \cite{web:johnnyfive}.
+Since it introduction in 1995~\cite{} there has been an increasing interest on JavaScript. Originally, this programming language was developed by Netscape Inc. to provide a lightweight interpreted language to captivate non-professional programmers. Later, thanks to a standardisation process started in 1996 with the first release~\cite{} and, that continues nowadays with the sixth version released in June 2015~\cite{}, Javascript has become one of the most popular programming languages on the Web.
 
+It is interesting to analyze the evolution of the language. Before the introduction of Ajax~\cite{} Javascript was principally used to do scripting in the Browser. In 2009 CommonJS~\cite{} and Node.js~\cite{web:nodejs} proposed an ecosystem for running Javascript outside the browser. With the rise of single-page web applications~\cite{} and source-to-source compilers~\cite{}, Javascript is now being used to power a large variety of projects. 
 
-With the rise of this programming language, a need has emerged for a pattern which can easily manage concurrent and distributed scenarios. Current approaches in the JavaScript world include the usage of callbacks and promises: the former of which has the unfortunate side-effect of forcing inversion of control, while the latter has no built-in solution for distributed computing.
+Despite the advances made by the javascript community, there are several issues that make developing software using Javascript harder. Working in Javascript is inconvenient because of its module system (only in ES6~\cite{}), its weak-typing, the verbose function syntax\footnote{Its support for closures is commonly noted as being one of JavaScript\rq{s} redeeming features.}, the late binding\footnote{Early binding allows for static verification of the existence of method-signature pairs}, the lack of static types, the \emph{this} behaviour, the equality/automatic conversion to boolean and the asynchronous programming heavily based on callbacks.

```

Comment:
Yes, it was a type. Anyway, it is clear that the workshop is about actors, but what I was questioning  in the introduction was:
1. that JavaScript has several problems 
2. that All the proposed solution does not tackle the problems related to asynchronous programming.

what do you think?